distributor/distributor.go

```diff
@@ -391,6 +392,31 @@ func (d *Distributor) LabelValuesForLabelName(ctx context.Context, labelName mod
 	return values, nil
 }
 
+// MetricsForLabelMatchers gets the metrics that match said matchers
+func (d *Distributor) MetricsForLabelMatchers(ctx context.Context, from, through model.Time, matcherSets ...metric.LabelMatchers) ([]metric.Metric, error) {
+	req, err := util.ToQueryRequest(from, through, matchers...)
+	if err != nil {
+		return nil, err
+	}
+	result := []metric.Metric{}
+	// TODO parallelise
+	for _, ingester := range d.cfg.Ring.GetAll() {
+		client, err := d.getClientFor(ingester)
+		if err != nil {
+			return nil, err
+		}
+
+		resp, err := client.MetricsForLabelMatchers(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+		for _, v := range resp.LabelValues {
```

Comment:
I guess this'll be `resp.Metrics`, but the corresponding `httpIngesterClient` change isn't in this PR yet.